nixos/doc/manual/configuration/x-windows.xml

```diff
@@ -230,27 +230,55 @@ xkb_symbols &quot;us-greek&quot;
   key &lt;LatZ&gt; { [ z, Z, Greek_zeta  ] };
 };
 </programlisting>
+  <note>
   <para>
-   To install the layout, the filepath, a description and the list of
-   languages must be given:
+   The longest layout name will be the width of the layout indicator
+   widget.
+  </para>
+  </note>
+  <para>
+   A minimal layout specification must include the following:
   </para>
 <programlisting>
 <xref linkend="opt-services.xserver.extraLayouts"/>.us-greek = {
   description = "US layout with alt-gr greek";
   languages   = [ "eng" ];
-  symbolsFile = /path/to/us-greek;
+  symbolsFile = /yourpath/symbols/us-greek;
 }
 </programlisting>
   <note>
   <para>
-   The name should match the one given to the
-   <literal>xkb_symbols</literal> block.
+   The name <literal>us-greek</literal> must match the one given to
+   the <literal>xkb_symbols</literal> block.
+  </para>
+  </note>
+  <note>
+  <para>
+   Make sure that your XKB file compiles and works fine!
   </para>
   </note>
   <para>
-   The layout should now be installed and ready to use: try it by
-   running <literal>setxkbmap us-greek</literal> and type
-   <literal>&lt;alt&gt;+a</literal>. To change the default the usual
+   Applying this customization requires rebuilding several packages,
+   and a broken XKB file can lead to the X session crashing at login.
+   Therefore, you're strongly advised to test your layout before
+   applying it:
+<screen>
+<prompt>$ </prompt>nix-shell -p xorg.xkbcomp
+<prompt>$ </prompt>setxkbmap -I/yourpath us-greek -print | xkbcomp -I/yourpath - $DISPLAY
+</screen>
+  </para>
+  <para>
+   You can inspect the predefined XKB files for examples:
+<screen>
+<prompt>$ </prompt>echo "$(nix-build --no-out-link '&lt;nixpkgs&gt;' -A xorg.xkeyboardconfig)/etc/X11/xkb/"
+</screen>
+  </para>
+  <para>
+   Once the configuration is applied, and you did a logout/login
+   cycle, the layout should be ready to use: try it by running
+   <literal>setxkbmap us-greek</literal> and type
+   <literal>&lt;alt&gt;+a</literal> (it may not get applied in your
```

Comment:
Why is this? As far as I know all X and even wayland applications use xkb.

Reply:
i don't know. but without using the gnome layout indicator widget to switch layouts, the new layout doesn't propagate to the gnome x terminal (but it does to e.g. Emacs).

and without the logout/login cycle, it doesn't show up in the list of available layouts. (which makes sense, because otherwise why would it be needed to rebuild the entire X and Gnome and apps locally?)